cmd/common.go

```diff
@@ -33,38 +32,38 @@ func SetStopCh(stopCh chan struct{}) {
 	stopChannel = stopCh
 }
 
-// checkWorkspace checks if workspace exists in Kong.
-func checkWorkspace(config utils.KongClientConfig) error {
+// workspaceExists checks if workspace exists in Kong.
+func workspaceExists(config utils.KongClientConfig) (bool, error) {
 
 	workspace := config.Workspace
 	if workspace == "" {
-		return nil
+		// default workspace always exists
+		return true, nil
 	}
 
-	client, err := utils.GetKongClient(config)
+	// remove workspace to be able to call top-level /workspaces endpoint
+	config.Workspace = ""
+	rootClient, err := utils.GetKongClient(config)
 	if err != nil {
-		return err
+		return false, err
 	}
 
-	req, err := http.NewRequest("GET",
-		utils.CleanAddress(config.Address)+"/workspaces/"+workspace,
-		nil)
-	if err != nil {
-		return err
-	}
-	resp, err := client.Do(nil, req, nil)
-	if resp != nil && resp.StatusCode == http.StatusNotFound {
-		return errors.Errorf("workspace '%v' does not exist in Kong, "+
-			"please create it before running decK.", workspace)
-	}
+	// Using ListAll() to get all workspaces instead of Get() to get just
+	// the one to make sure there is no underlying error - Get() does not
+	// expose the response.
+	workspaces, err := rootClient.Workspaces.ListAll(nil)
```

Comment:
> the one to make sure there is no underlying error

Yes, this is a limitation and go-kong needs to improve to fix this problem but I still think we should use Get() instead. You can use [IsNotFoundErr()](https://godoc.org/github.com/hbagdi/go-kong/kong#IsNotFoundErr) to check if it was a 404 or not.

If there is an underlying error, return it.

Reply:
Makes sense, and done.